storage/objects/stream_file_upload.go

```diff
@@ -43,8 +43,21 @@ func streamFileUpload(w io.Writer, bucket, object string) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
 	defer cancel()
 
+	o := client.Bucket(bucket).Object(object)
+
+	// Add preconditions so that the upload may safely retry on transient errors.
+	// In this code sample, we assume the object does not yet exist.
```

Comment:
Suggestion:

Replace "In this code sample..." with "The following code sample sets a `Match` precondition to 0. The request to upload an object is aborted if the live object already exists in your storage bucket."

Reply:
I'm concerned about this wording since it may be confusing to users why we use this condition instead of GenerationMatch and/or may make them try to use `GenerationMatch: 0` instead (which does nothing, as we do say in the go docs). On the other hand, this is accurate and may help users understand that the `DoesNotExist` precondition is equivalent to setting a generation match but for non-existing objects.

Reply:
What do you propose as a change to the existing wording?